source/common/stats/thread_local_store.cc

```diff
@@ -197,12 +197,9 @@ void ThreadLocalStoreImpl::mergeHistograms(PostMergeCb merge_complete_cb) {
     merge_in_progress_ = true;
     tls_->runOnAllThreads(
         [this]() -> void {
-          for (const auto& scope : tls_->getTyped<TlsCache>().scope_cache_) {
-            const TlsCacheEntry& tls_cache_entry = scope.second;
-            for (const auto& name_histogram_pair : tls_cache_entry.histograms_) {
-              const TlsHistogramSharedPtr& tls_hist = name_histogram_pair.second;
-              tls_hist->beginMerge();
-            }
+          for (const auto& id_hist : tls_->getTyped<TlsCache>().histogram_cache_) {
```

Comment:
We no longer need to iterate through all scopes because we have all histograms an an independent cache, is that right?

Reply:
correct.